indexing-service/src/main/java/io/druid/indexing/overlord/RemoteTaskRunner.java

```diff
@@ -872,4 +873,58 @@ private void taskComplete(
     // Notify interested parties
     taskRunnerWorkItem.setResult(taskStatus);
   }
+
+  public Collection<ZkWorker> findLazyworkers(Predicate<ZkWorker> isLazyWorker, int excessWorkers)
+  {
+    // status lock is used to prevent any tasks being assigned to the worker while we mark it lazy
+    synchronized (statusLock) {
+      Iterator<String> iterator = zkWorkers.keySet().iterator();
+      while (iterator.hasNext()) {
+        String worker = iterator.next();
+        ZkWorker zkWorker = zkWorkers.get(worker);
+        try {
+          if (getAssignedTasks(zkWorker.getWorker()).isEmpty() && isLazyWorker.apply(zkWorker)) {
+            log.info("Adding Worker[%s] to lazySet!", zkWorker.getWorker().getHost());
+            lazyWorkers.put(worker, zkWorker);
+            iterator.remove();
```

Comment:
It seems like we are relying on both zkWorkers and lazyWorkers to know which ones are available, which seems redundant. If we already added a node to lazyWorkers, is there a reason to remove it from zkWorkers?